ext/asm_event.c

```diff
@@ -18,9 +18,8 @@ DDTRACE_PUBLIC void ddtrace_emit_asm_event() {
     } else {
         DDTRACE_G(asm_event_emitted) = true;
     }
-    if (!get_DD_APM_TRACING_ENABLED()) {
-        ddtrace_set_priority_sampling_on_root(PRIORITY_SAMPLING_USER_KEEP, DD_MECHANISM_ASM);
-    }
+
+    ddtrace_set_priority_sampling_on_root(PRIORITY_SAMPLING_USER_KEEP, get_DD_APM_TRACING_ENABLED() ? DD_MECHANISM_MANUAL : DD_MECHANISM_ASM);
```

Comment:
Are you sure it makes sense to use `DD_MECHANISM_MANUAL` considering it sets `_dd.p.dm` to `-4` (which isn't accurate)?

Reply:
After an offline conversation, the conclusion is that we should just change this to be `DD_MECHANISM_ASM` when we set manual keep.

Reply:
So this line will change to the following if I understood correct
```
ddtrace_set_priority_sampling_on_root(PRIORITY_SAMPLING_USER_KEEP,  DD_MECHANISM_ASM);
```

Reply:
Done